internal/controller/metrics/metrics.go

```diff
@@ -173,3 +185,32 @@ func (m metricsImpl) SetKubeconfigExpiration(secret corev1.Secret, rotationPerio
 		}
 	}
 }
+
+func (m metricsImpl) SetPendingStateDuration(operation string, runtime v1.Runtime) {
+	runtimeID := runtime.GetLabels()[RuntimeIDLabel]
+
+	getDuration := func() time.Duration {
+		if runtime.Status.State == v1.RuntimeStatePending {
+			condition := meta.FindStatusCondition(runtime.Status.Conditions, string(v1.ConditionTypeRuntimeProvisioned))
+			return time.Since(condition.LastTransitionTime.Time)
+		}
+
+		if runtime.Status.State == v1.RuntimeStateTerminating {
+			return time.Since(runtime.DeletionTimestamp.Time)
+		}
+
+		return 0
+	}
+
+	if runtimeID != "" {
```

Comment:
Reverse `if` for early return. With early return, could we turn `getDuration` lambda into just body code of this method?